File: app/src/main/java/com/example/msimple/adapter/ProgramAdapter.java
package com.example.msimple.adapter;

import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.msimple.Delegate.OnClickItem;
import com.example.msimple.R;
import com.example.msimple.data.vos.ProgramVO;
import com.example.msimple.view.holders.BaseHolder;
import com.example.msimple.view.holders.ProgramHolder;

public class ProgramAdapter extends BaseAdapter<ProgramHolder, ProgramVO> {


    OnClickItem onClickItem;
    public ProgramAdapter(OnClickItem clickItem) {
        onClickItem = clickItem;
    }

    @NonNull
    @Override
    public ProgramHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        View view = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.item_view_program,viewGroup,false);
        return new ProgramHolder(view,onClickItem);
    }
}

File: app/src/main/java/com/example/msimple/adapter/ProgramAdapter.kt
package com.example.msimple.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.example.msimple.delegate.OnClickItem
import com.example.msimple.R
import com.example.msimple.data.vos.ProgramVO
import com.example.msimple.view.holders.ProgramHolder

class ProgramAdapter(internal var onClickItem: OnClickItem) : BaseAdapter<ProgramHolder, ProgramVO>() {

    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ProgramHolder {
        val view = LayoutInflater.from(viewGroup.context).inflate(R.layout.item_view_program, viewGroup, false)
        return ProgramHolder(view, onClickItem)
    }
}
